src/hooks/useCurrentMetric.ts

```diff
@@ -19,7 +19,7 @@ export function useCurrentMetric() {
 
   let currentMetric = queryParamsMetric as Metric
 
-  if (!queryParamsMetric || !['eau', 'carbone'].includes(queryParamsMetric)) {
+  if (!queryParamsMetric || !metrics.includes(queryParamsMetric as Metric)) {
```

Comment:
You can cast `queryParamsMetric` as `Metric` once and for all on line 11

Reply:
Unless queryParamsMetric is empty no?

Reply:
Then `queryParamsMetric` should be `Metric | undefined`

Reply:
Deal